test/middleware.cjs

```diff
@@ -281,6 +281,33 @@ describe('middleware', () => {
       }
     });
 
+    // Addresses: https://github.com/yargs/yargs/issues/2124
+    // This test will fail if the result of async middleware is not treated like a promise
+    it('treats result of async middleware as promise', done => {
```

Comment:
I would be tempted to write a more specific test to #2124 

"Using the strict option in combination with an async middleware, which is applied before the validation, does not result in unknown command"

Reply:
I simplified the test. I kept the `done` logic, because Mocha will pass this test before the unhandled promise would cause it to fail. If there's a regression, this test will fail because `done` will get called twice, and one of the executions will have an error passed as an argument